### PR TITLE
Bluetooth: GATT: Rework BT_GATT_CHARACTERISTIC

### PR DESCRIPTION
**Pull request by Vudentz**
These changes ensure the Characteristic Value is always declared with the same UUID as indicated in the characteristic declaration.

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]
Looks ok to me

**Review by carlescufi — APPROVED** [review]
Thank you @Vudentz. This more than solves my concerns there with the current macros.

File: samples/bluetooth/gatt/hrs.c [hunk 1]
```diff
@@ -43,17 +43,13 @@ static ssize_t read_blsc(struct bt_conn *conn, const struct bt_gatt_attr *attr,
 /* Heart Rate Service Declaration */
 static struct bt_gatt_attr attrs[] = {
 	BT_GATT_PRIMARY_SERVICE(BT_UUID_HRS),
-	BT_GATT_CHARACTERISTIC(BT_UUID_HRS_MEASUREMENT, BT_GATT_CHRC_NOTIFY),
-	BT_GATT_DESCRIPTOR(BT_UUID_HRS_MEASUREMENT, BT_GATT_PERM_READ, NULL,
-			   NULL, NULL),
+	BT_GATT_CHARACTERISTIC(BT_UUID_HRS_MEASUREMENT, BT_GATT_CHRC_NOTIFY,
+			       BT_GATT_PERM_READ, NULL, NULL, NULL),
```
Comment (carlescufi) [inline samples/bluetooth/gatt/hrs.c]: This should not have `PERM_READ`

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Review by asbjornsabo — COMMENTED** [review]
I am in favour of this change.  The previous behaviour, using DESCRIPTOR to add the value of a characteristic, confused me.  Neither did this previous use of DESCRIPTOR correspond to what a descriptor really is.

**Comment by Vudentz** [comment]
Patches updated.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**